packages/block-editor/src/components/block-selection-clearer/index.js

```diff
@@ -10,22 +10,18 @@ import { useRefEffect } from '@wordpress/compose';
 import { store as blockEditorStore } from '../../store';
 
 export function useBlockSelectionClearer( onlySelfClicks = false ) {
-	const hasSelection = useSelect( ( select ) => {
-		const { hasSelectedBlock, hasMultiSelection } = select(
-			blockEditorStore
-		);
-
-		return hasSelectedBlock() || hasMultiSelection();
-	} );
+	const { hasSelectedBlock, hasMultiSelection } = useSelect(
```

Comment:
Why this change was needed?

Reply:
With #31078, any selector calls where the result is used in an event callback should be called only when necessary in the callback instead, and in addition it will always be fresh. Not sure why the component didn't re-render before, it seems worth investigating.